sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/simpleCosting.scala

```diff
@@ -38,7 +38,7 @@ case class SimpleCost(value: Long) extends Cost {
  * A simple implementation of [[CostEvaluator]], which counts the number of
  * [[ShuffleExchangeLike]] nodes in the plan.
  */
-object SimpleCostEvaluator extends CostEvaluator {
+case class SimpleCostEvaluator() extends CostEvaluator {
```

Comment:
This can still be an object, if we follow https://github.com/apache/spark/pull/32944/files#r662513062

Reply:
@cloud-fan - yeah, updated.